File: solutions/widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.py
# sort nums 
# iterate thru x array
#     get difference between next and current num
#     update width if current difference is greater than previous
# return width

class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        points.sort(key=lambda point:point[0])
        
        width = 0
        
        for i in range(len(points) - 1):
            curr = points[i]
            next = points[i + 1]
            
            width = max(next[0] - curr[0], width)
                                       
        return width

File: solutions/widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */

// track nums for x
// iterate thru points arr
    // grab x and push into x array
// sort nums 
// iterate thru x array
    // get difference between next and current num
    // update width if current difference is greater than previous
// return width

var maxWidthOfVerticalArea = function(points) {
    let xCoords = [];
    
    for (let i = 0; i < points.length; i++) {
        let point = points[i];
        
        xCoords.push(point[0]);
    }
    
    xCoords.sort((a, b) => a - b);
    
    let width = 0;

    for (let i = 0; i < xCoords.length - 1; i++) {
        let next = xCoords[i + 1];
        let curr = xCoords[i];
        
        width = Math.max(width, next - curr);
    }
    
    return width;
};
